src/Okta.Idx.Sdk/IdxConfiguration.cs

```diff
@@ -41,6 +41,16 @@ public class IdxConfiguration
         /// </summary>
         public string RedirectUri { get; set; }
 
+        /// <summary>
+        /// Gets or sets the acrvalue URI.
+        /// </summary>
+        public string AcrValue { get; set; }

```

Comment:
I suggest removing AcrValue and LoginHint from the configuration and making them optional parameters of the AuthenticateAsync method.

Reply:
InteractAsync gets called from multiple places in IdxClient.cs and InteractAsync gets all the data for payload from IdxConfiguration so from that point of view it feels like having this in IdxConfiguration is seems to be more natural. Also, adding this as optional parameter in AuthenticateAsync will break the public API signature (or I can create an overload). In future if other APIs also need these parameters then it will be another change in SDK. I feel like adding these as optional parameters is more intrusive change.
But I will lean towards your recommendation and ok either way.

Reply:
Please, help me understand the purpose of ACR values and login_hint.
The way I see this is that `login_hint`, for example, is more a request parameter than a client configuration. I imagine that for the same client configuration (client credentials, etc.) you can still want a different `login_hint` for every user that calls the Authenticate method. The IdxConfiguration should contain attributes that won't change once you have instantiated a client, no matter what user tries to authenticate. Does it make sense?

If this makes sense, you can either add a new property to `AuthenticationOptions` and accept null values to avoid breaking changes or add support for request params in the `RequestContext` class, which (again) should be optional to avoid breaking changes.

ACR Values seem to be a different case, and it might make sense to keep it in the IdxConfiguration since, per OIDC definition, this establishes an agreement between the client and the IDP. Can this change per user? 

Let me know if this makes sense.

Reply:
acr_values will not change per user.
I agree with login_hint not being part of IdxConfiguration.
I will update the PR and share it shortly.
thank you for explianing.